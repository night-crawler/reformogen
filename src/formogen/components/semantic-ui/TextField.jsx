import React from 'react';
import PropTypes from 'prop-types';

import _ from 'lodash';

import { Form } from 'semantic-ui-react';
import { errorsType, layoutOptsType } from '../fieldPropTypes';
import Label from './Label';
import { MessageList } from './MiscComponents';

TextField.propTypes = {
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    value: PropTypes.string,
    max_length: PropTypes.number,
    help_text: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    errors: errorsType,

    required: PropTypes.bool,
    editable: PropTypes.bool,

    helpTextOnHover: PropTypes.bool,
    layoutOpts: layoutOptsType,

    updateProps: PropTypes.func,
    onChange: PropTypes.func,
};
export default function TextField(props) {
    let _props = {
        name: props.name,
        value: props.value,
        onChange: props.onChange,
        placeholder: props.placeholder,
    };

    if (_.isFunction(props.updateProps)) {
        _props = props.updateProps(_props, props);
    }

    return (
        <Form.Field
            required={ props.required }
            disabled={ !props.editable }
            width={ props.layoutOpts.width }
            error={ !_.isEmpty(props.errors) }
        >
            <Label { ...props } />
            <Form.TextArea { ..._props } />
            { !props.helpTextOnHover ? <span className='help-text'>{ props.help_text }</span> : '' }
            <MessageList messages={ props.errors } />
        </Form.Field>
    );
}
