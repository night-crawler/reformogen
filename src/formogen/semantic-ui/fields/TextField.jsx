import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Form } from 'semantic-ui-react';

import { errorsType, layoutOptsType } from '../../fieldPropTypes';
import Label from '../common/Label';
import { ErrorsList } from '../common/ErrorsList';

TextField.propTypes = {
  type: PropTypes.string.isRequired,

  layoutOpts: layoutOptsType,

  name: PropTypes.string.isRequired,
  verbose_name: PropTypes.string.isRequired,
  help_text: PropTypes.string.isRequired,
  errors: errorsType,

  max_length: PropTypes.number,

  editable: PropTypes.bool,
  required: PropTypes.bool,

  value: PropTypes.string,

  placeholder: PropTypes.string,

  helpTextOnHover: PropTypes.bool,

  updateProps: PropTypes.func,

  onChange: PropTypes.func,
};
export function TextField(props) {
  let _props = {
    name: props.name,
    value: props.value,
    onChange: props.onChange,
    placeholder: props.placeholder,
  };

  if (_.isFunction(props.updateProps)) {
    _props = props.updateProps(_props, props);
  }

  return (
    <Form.Field
      required={ props.required }
      disabled={ !props.editable }
      width={ props.layoutOpts.width }
      error={ !_.isEmpty(props.errors) }
    >
      <Label { ...props } />
      <Form.TextArea { ..._props } />
      { !props.helpTextOnHover
        ? <span className='help-text'>{ props.help_text }</span>
        : ''
      }
      <ErrorsList messages={ props.errors } />
    </Form.Field>
  );
}
