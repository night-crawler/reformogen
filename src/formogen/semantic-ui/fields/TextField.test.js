import TextField from './TextField';


it('should render TextField', () => {
  const wrapper = shallow(
    <TextField
      type='TextField'
      name='text_field_name'
      verbose_name='verbose name'
      help_text='help text'
      layoutOpts={ { width: 1 } }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
