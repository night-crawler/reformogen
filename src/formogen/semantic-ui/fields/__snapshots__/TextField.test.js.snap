// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextField 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextField
    help_text="help text"
    layoutOpts={
      Object {
        "width": 1,
      }
    }
    name="text_field_name"
    type="TextField"
    verbose_name="verbose name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <FieldLabel
          help_text="help text"
          layoutOpts={
            Object {
              "width": 1,
            }
          }
          name="text_field_name"
          type="TextField"
          verbose_name="verbose name"
        />,
        <FormTextArea
          as={[Function]}
          control={[Function]}
          name="text_field_name"
        />,
        <span
          className="help-text"
        >
          help text
        </span>,
        <ErrorsList
          color="red"
          header="Errors"
        />,
      ],
      "disabled": true,
      "error": false,
      "required": undefined,
      "width": 1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "help_text": "help text",
          "layoutOpts": Object {
            "width": 1,
          },
          "name": "text_field_name",
          "type": "TextField",
          "verbose_name": "verbose name",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": [Function],
          "control": [Function],
          "name": "text_field_name",
          "onChange": undefined,
          "placeholder": undefined,
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "help text",
          "className": "help-text",
        },
        "ref": null,
        "rendered": "help text",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "red",
          "header": "Errors",
          "messages": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <FieldLabel
            help_text="help text"
            layoutOpts={
              Object {
                "width": 1,
              }
            }
            name="text_field_name"
            type="TextField"
            verbose_name="verbose name"
          />,
          <FormTextArea
            as={[Function]}
            control={[Function]}
            name="text_field_name"
          />,
          <span
            className="help-text"
          >
            help text
          </span>,
          <ErrorsList
            color="red"
            header="Errors"
          />,
        ],
        "disabled": true,
        "error": false,
        "required": undefined,
        "width": 1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "help_text": "help text",
            "layoutOpts": Object {
              "width": 1,
            },
            "name": "text_field_name",
            "type": "TextField",
            "verbose_name": "verbose name",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": [Function],
            "control": [Function],
            "name": "text_field_name",
            "onChange": undefined,
            "placeholder": undefined,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "help text",
            "className": "help-text",
          },
          "ref": null,
          "rendered": "help text",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "red",
            "header": "Errors",
            "messages": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
